spec/core.md

```diff
@@ -256,6 +256,249 @@ Finally, concatenate the prefix "did:peer:3" with the computed and encoded hash:
+    4. Consider this value the `hash`
+3. Construct the did by concatenating the values as follows:
+
+        did:peer:4{{hash}}:{{encoded document}}
+
+Here is an example long form DID made from the input example above:
+
+```
+did:peer:4zQmNsz8npvrAyj983LTownQhp3PmGVGzMYrhBRGfig6rZ6P:z3U8mHxj7WY8i5EQD5rmkfPaDTws9dw5NRERbsFhsDFh5kvAEAY1afMFTWWbZT2d7TjCYtGBt47e1WpYoXYKSoEjxQX9qhNSw2ujMPiHn5rVyVCwQH1kBLzs95n37MoD6ikVDd29fkJoZaBrGhNaxcFiW9b1N85NWGQ9vs1LEMxsqt5ZHiFXGp1tMU85Da4VETb7veFZjvxRsadgbgeK1tFG83c6SSyDjoVzQbagXJ44kZXaMMmbs4Kk6VJszrCM1CUHPk7kFgHmojWU8kFAmPE88oesxBW5Wdc5cCNB6cB4QzRGuYrkK15hMdoFWq4kVHJn1xbruCzJ7Mb9f2rKUF6KTiUbTYpHuo4Kbnu26tJKQ9D7xHkAM2N3ZPN3eRkXBefKme5mLhGjRgXm6fAZiCHuK4dMyg4Bd9HDXiy8vSdY8cyZnyuJdPsjq5FRvRD92cFNtJZBJJwRQu6WiwKhTL9jELwwGfU2jukeESmARHjpQRTkXhtyG5NHDwj3Yx9CsbyBR5xdGsB3raA8JiMP4nAsbZhfXiBErBUx4MwYRnBDZERZztPjJWJniyKVG6hfoBokzEtkZt6gYMh1tpjsBAcSVw4C9H7o7QrY3mW6DjSufDdHSdWPVJjfHgRzxUM218CSiEwqEctqxJP9fc2FVSDxai7JUnroVzgYzhb62S4ueLGKM83abkd3Fm5NeSuewPRbgwETTLvknz1Wq1G4qygq75Fp3Kr21qknM2tsgrkwyprYR9ZTK5YzY5sHCwNP14VXZeX24QdSfevspNdvFtFtiDq6dUufmy5bKeLdHxx7Mpb7vFToU8bk9zZNUkcgXvX12U6iT1zLEyszTS6B3csHRr1HmvLUgEQKfd2aWjV2ScktEBsjZRHWdWuxQRcs85sF92kW2fVX7k1EGAwYGsnr6Wf9Q7jkM7SgJM5WJ1rHHsxKXwj8j11QmwXRcgREVEJphWdj87tpsC36A4rfEYhyJDw13UB68JgoK544NbsA
+```
+
+To construct the short form, simply omit the `:{{encoded document}}` from the end.
+
+Here is an example short form DID for the long form above:
+
+```
+did:peer:4zQmNsz8npvrAyj983LTownQhp3PmGVGzMYrhBRGfig6rZ6P
+```
+
+##### Resolving a DID
+
+###### Long form
+
+Resolving a long form `did:peer:4` document is done by decoding the document from the DID and &#34;contextualizing&#34; the document with the DID.
+
+To decode the document:
+
+1. Extract the `encoded document` portion of the DID
+2. Verify the hash over the `encoded document` by extracting the `hash` portion of the DID and comparing it against the result of following step 2 (&#34;Hash the document&#34;) above to recreate the hash.
+3. Perform the inverse of step 1 (&#34;Encode the document&#34;) to get the decoded document
+
+To &#34;contextualize&#34; a document:
+
+1. Take the decoded document (which should look identical to the input example above)
+2. Add `id` at the root of the document and set it to the DID
+3. Add `alsoKnownAs` at the root of the document and set it to a list, if not already present, and append the short form of the DID
+4. For each verification method (declared in the `verificationMethod` section or embedded in a verification relationship like `authentication`):
+    - If `controller` is not set, set `controller` to the DID
+
+&gt; Note: Implementations may turn relative references in the document into absolute references by prepending the reference with the DID. This is not recommended due to length but this is an implementation detail that should not affect usage of the resolved document. Both relative and absolute references are valid within DID Documents.
```

Comment:
```suggestion
> Note: Implementations may turn relative references in the document into absolute references by prepending the reference with the DID. This is not recommended due to length but this is an implementation detail that should not affect usage of the resolved document. Both relative and absolute references are valid within DID Documents.
```

Reply:
Maybe this one should be a `::: note ::::` style note?

Reply:
It's cool, though Github seems to favor `> [!note]` syntax, but it is less universally supported.